docs/releasing/README.md

```diff
@@ -0,0 +1,182 @@
+# Releasing
+
+## When to release
+
+- When a [Subnet-EVM](https://github.com/ava-labs/avalanchego/releases) release is made
+- If there is a significant code change or bugfix in this repository
+
+## Procedure
+
+In this section, we create a release `v0.4.0`. We therefore assign these environment variables to simplify copying instructions:
+
+```bash
+export VERSION=v0.4.0
+export GITHUB_USER=username
+```
+
+`GITHUB_USER` is the username of the person creating the release. This is simply used to prefix branches.
+
+1. Create your branch, usually from the tip of the `main` branch:
+
+    ```bash
+    git fetch origin main:main
+    git checkout main
+    git checkout -b "$GITHUB_USER/releases/$VERSION"
+    ```
+
+1. Modify the [plugin/main.go](../../plugin/main.go) `Version` global string constant and set it to the desired `$VERSION`.
+1. Upgrade the [subnet-evm Go dependency](https://github.com/ava-labs/subnet-evm/releases), which will likely also upgrade the AvalancheGo dependency version:
+
+    ```bash
+    go get github.com/ava-labs/subnet-evm
+    go mod tidy
+    ```
+
+1. Get the Go version from the `go.mod` file:
+
+    ```bash
+    cat go.mod | grep -oE "^go 1\.\d+" | grep -oE "1\.\d+"
+    ```
+
+    and update the [.devcontainer/devcontainer.json](../../.devcontainer/devcontainer.json) file with the Go version at `.features.["ghcr.io/devcontainers/features/go:1.version"].version`, for example:
+
+    ```json
+    {
+        "features": {
+            "ghcr.io/devcontainers/features/go:1": {"version": 1.23},
+        }
+    }
+    ```
+
+1. Get the AvalancheGo version from the `go.mod` file:
+
+    ```bash
+    go list -m all | grep github.com/ava-labs/avalanchego | awk '{print $2}'
+    ```
+
+    and update the [.devcontainer/devcontainer.json](../../.devcontainer/devcontainer.json) file with it at `.build.args.AVALANCHEGO_VERSION`, for example:
+
+    ```json
+    {
+        "build": {
+            "args": {
+                "AVALANCHEGO_VERSION": "v1.13.0"
+            }
+        },
+    }
+    ```
+
+1. Add an entry in the object in [compatibility.json](../../compatibility.json), adding the target release `$VERSION` as key and the AvalancheGo RPC chain VM protocol version as value, to the `"rpcChainVMProtocolVersion"` JSON object. For example, we would add:
+
+    ```json
+    "v0.4.0": 39,
+    ```
+
+    💁 If you are unsure about the RPC chain VM protocol version, set the version to `0`, for exampple `"v0.4.0": 0`, and then run:
+
+    ```bash
+    go test -run ^TestCompatibility$ github.com/ava-labs/precompile-evm/plugin
+    ```
+
+    This will fail with an error similar to:
+
+    ```text
+    compatibility.json has precompile-evm version v0.4.0 stated as compatible with RPC chain VM protocol version 0 but AvalancheGo protocol version is 39
+    ```
+
+    This message can help you figure out what the correct RPC chain VM protocol version (here `39`) has to be in compatibility.json for your current release.
```

Comment:
We should probably also mention where/how to get this version (i.e in avalanchego release notes or the file itself)